gwyfile/objects.py

```diff
@@ -348,7 +348,7 @@ def component_from_buffer(buf, return_size=False):
         }
         dtype = typelookup[typecode]
         pos, endpos = endpos, endpos + dtype.itemsize * numitems
-        data = np.fromstring(buf[pos:endpos], dtype=dtype)
+        data = np.frombuffer(buf[pos:endpos], dtype=dtype)
```

Comment:
I think there might be a subtle change in behavior ([numpy docs](https://numpy.org/doc/stable/reference/generated/numpy.frombuffer.html)):
> This function creates a view into the original object. This should be safe in general, but it may make sense to copy the result when the original object is mutable or untrusted.

After the change, `GwyObject.frombuffer` returns only a view of the data whereas before data was copied. If the input buffer is modified, the output may change as well.

However, that's a niche use case and adopting the behavior of `numpy.frombuffer` makes a lot of sense. It's better to leave the copy to the user.

So, all green from my side. Together with dropping Python 2 support that justifies release 0.3.